File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node



class Rectangle {
	constructor (w, h) {
		if (w <= 0 || h <= 0) {
			return {};
		} else {
			this.width = w;
			this.height = h;
		}
	}

	print () {
		for (let i = 0; i <= this.height; i++) {
			let line = '';
			for (let j = 0; j <= this.width; j++) {
				line += 'X';
			}
			console.log(line);
		}
	}
}

const r1 = new Rectangle(2, 3);
r1.print();

const r2 = new Rectangle(10, 5);
r2.print();

File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""Defines a class module"""


class Rectangle:
    """Represent a class"""
    def __init__(self, width=0, height=0):
        """Initialize a class.
        Parameteres:
            width (int)
            height (int)
        Raises:
            TypeError: number must be integer.
            ValueError: number must be >= 0.
        Return:
            Area (int):
            perimeter (int):
        """
        self.__width = width
        self.__height = height

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        return self.__width * self.__height

    def perimeter(self):
        if self.__width == 0 and self.__height == 0:
            return 0
        return 2 * (self.__width + self.__height)

    def __str__(self):
        if self.__width == 0 or self.__height == 0:
            return ""
        result = ""
        for i in range(self.__height):
            result += "#" * self.__width + "\n"
        return result
        
    def __repr__(self):
        return "Rectangle({}, {})".format(self.__width, self.__height)
